test/utils/padder/padder.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package padder
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrtv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	"github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	nrtutil "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+// This package allows to control the amount of available resources under the nodes.
+// it's doing that by "padding" the nodes with guaranteed pods until node reaches the desired amount of available resources.
+//
+// Usage example:
+// pad := New("test-ns")
+// pad.Nodes(2).UntilAvailableReach(corev1.ResourceCPU, "3").UntilAvailableReach(corev1.ResourceMemory, "100Mi").Pad()
+// or
+// rl := corev1.ResourceList {
+//		corev1.ResourceCPU: "10",
+//		corev1.ResourceMemory: "100Mi"
+//	}
+// pad.Nodes(2).UntilAvailableReachWithResourceList(rl).Pad()
+
+type Padder struct {
+	// Client defines the API client to run CRUD operations, that will be used for testing
+	Client    client.Client
+	Namespace string
+	*PadRequest
+}
+
+type PadRequest struct {
+	nNodes    int
+	resources corev1.ResourceList
+}
+
+// New return new padder object
+// ns is where padding pods will get deployed
+func New(ns string) (*Padder, error) {
+	padder := &Padder{
+		Namespace: ns,
+	}
+
+	if !e2eclient.ClientsEnabled {
+		return nil, fmt.Errorf("clients not enabled")
+	}
+
+	padder.Client = e2eclient.Client
+	padder.PadRequest = &PadRequest{resources: make(corev1.ResourceList)}
+	return padder, nil
+}
+
+// Nodes number of nodes to pad
+func (p *Padder) Nodes(n int) *Padder {
+	p.PadRequest.nNodes = n
+	return p
+}
+
+// UntilAvailableReach will pad pods into nodes until reach the desired available resources
+func (p *Padder) UntilAvailableReach(resName corev1.ResourceName, quantity string) *Padder {
+	quan := resource.MustParse(quantity)
```

Comment:
panicing here is not a great UX, though.

Reply:
Agree. 
But:
1. It's a testing utility that is used for testing purposes only, so we basically know the string is valid.
2. If we need to return an error here, it kinda breaks the fluent  API design.

Reply:
Valid points. Let's keep like it is (circling a bit back to "not a huge fan of the fluent api design", but still)